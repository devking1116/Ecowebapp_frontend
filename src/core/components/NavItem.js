/* @flow */
import React from 'react';
import reactCSS from 'reactcss';
import { Button, Icon } from 'semantic-ui-react';

type Props = {
  label: string,
  active?: bool,
};

/**
 * Component that works as a drop down menu for the navigation bar.
 */
export default function NavItem({ label, active, ...otherProps }: Props): React.Element<*> {
  const styles = reactCSS({
    'default': {
      icon: {
        marginLeft: 15,
      },
    },
  });

  return (
    <Button color={active? 'blue': undefined} inverted={!active} active={active} {...otherProps}>
      {label}
      <Icon style={styles.icon} name='angle down' />
    </Button>
  );
}
