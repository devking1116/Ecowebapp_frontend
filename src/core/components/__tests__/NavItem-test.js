import React from 'react';
import renderer from 'react-test-renderer';

import NavItem from '../NavItem';

test('it displays label next to the arrow.', () => {
  const component = renderer.create(
    <NavItem label="Test label" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
