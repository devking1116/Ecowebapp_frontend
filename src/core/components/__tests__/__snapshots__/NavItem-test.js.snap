exports[`test it displays label next to the arrow. 1`] = `
<button
  className="ui inverted button"
  onClick={[Function]}
  tabIndex={undefined}>
  Test label
  <i
    aria-hidden="true"
    className="angle down icon"
    style={
      Object {
        "marginLeft": 15,
      }
    } />
</button>
`;
